lib/resource-serializer.js

```diff
@@ -1,9 +1,22 @@
 'use strict';
+var _ = require('lodash');
 var SerializerUtils = require('./serializer-utils');
 
 function ResourceSerializer(collectionName, record, opts) {
   var payload = { data: {} };
 
+  function getLinks(links) {
+    return _.mapValues(links, function (value) {
+      if (_.isFunction(value)) {
+        return value(records);
+      } else {
+        return value;
+      }
+    });
+  }
+
+  if (opts.topLevelLinks) { payload.links = getLinks(opts.topLevelLinks); }
+
```

Comment:
I think it will be better if there's no code duplication here between the `getLinks` function in the `collection-serializer` and the `getLinks` function (which is the same) in the `resource-serializer`.